link.go

```diff
@@ -111,11 +116,40 @@ func (link *ToxicLink) UpdateToxic(toxic *toxics.ToxicWrapper) {
 func (link *ToxicLink) RemoveToxic(toxic *toxics.ToxicWrapper) {
 	i := toxic.Index
 
-	// Interrupt the last toxic so that the target's buffer is empty
-	if link.stubs[i-1].InterruptToxic() && link.stubs[i].InterruptToxic() {
+	if link.stubs[i].InterruptToxic() {
+		stop := make(chan bool)
+		// Interrupt the previous toxic to update its output
+		go func() {
+			stop <- link.stubs[i-1].InterruptToxic()
+		}()
+
+		// Unblock the previous toxic if it is trying to flush
+		// If the previous toxic is closed, continue flusing until we reach the end.
+		interrupted := false
+		stopped := false
+		for !interrupted {
+			select {
+			case interrupted = <-stop:
+				stopped = true
+			case tmp := <-link.stubs[i].Input:
+				if tmp == nil {
```

Comment:
Do you really need the case above if you have this case? Or does it only return the end of the stream as `nil` once?